app/models/patient.rb

```diff
@@ -246,9 +246,9 @@ def notes_count
   end
 
   def has_special_circumstances
-    has_circumstance = 0
+    has_circumstance = false
```

Comment:
This is great! I think we got tripped up by the handling here and what ruby considers truthy.

@ChrisBarthol what do you think about simplifying this a little? The variable assignment and then boolean casting is kinda tricky. Maybe something like this would work?

`special_circumstances.map { |x| x.present? }.any?`

That would cast each of the special circumstances to true or false, and then return true if there are any trues or false otherwise. Ex:

```
irb(main):001:0> [false, true, true].any? 
=> true
irb(main):002:0> [false, false, false].any? 
=> false
```

what do you think?

Reply:
Great catch on the simplification!  Switched to map and added a few extra test cases.